scipy/stats/morestats.py

```diff
@@ -2706,17 +2706,27 @@ def circmean(samples, high=2*pi, low=0, axis=None):
     Returns
     -------
     circmean : float
+        nan value if no mean can be calculated (e.g., uniform 
+            circular distribution)
```

Comment:
Uhm, I don't think we need to revert anything. Your original PR, https://github.com/scipy/scipy/pull/5662, is a good one, we should keep it. A small bug did slip past, and that's OK. All I'm saying is that if you fix the bug in this PR --- so that what worked in 0.17.1 keeps working with this PR --- this is enough for this PR. Any further improvements should go to a separate one, that's all.